src/main/java/net/fabricmc/mappingio/format/FeatureSetBuilder.java

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright (c) 2024 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.mappingio.format;
+
+import java.util.function.Consumer;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.fabricmc.mappingio.format.FeatureSet.DescSupport;
+import net.fabricmc.mappingio.format.FeatureSet.ElementCommentSupport;
+import net.fabricmc.mappingio.format.FeatureSet.FeaturePresence;
+import net.fabricmc.mappingio.format.FeatureSet.LocalSupport;
+import net.fabricmc.mappingio.format.FeatureSet.MemberSupport;
+import net.fabricmc.mappingio.format.FeatureSet.MetadataSupport;
+import net.fabricmc.mappingio.format.FeatureSet.NameSupport;
+import net.fabricmc.mappingio.format.FeatureSetImpl.DescSupportImpl;
+import net.fabricmc.mappingio.format.FeatureSetImpl.LocalSupportImpl;
+import net.fabricmc.mappingio.format.FeatureSetImpl.MemberSupportImpl;
+import net.fabricmc.mappingio.format.FeatureSetImpl.NameSupportImpl;
+
+@ApiStatus.Experimental
+public class FeatureSetBuilder {
+	public static FeatureSetBuilder create() {
```

Comment:
Since your subbuilders are using `Consumer<SubBuilder>`, should the root builder also be like:
```java
public static FeatureSet build(Consumer<FeatureSetBuilder> featureApplier)
```
and another overload with an existing `FeatureSet`?

Reply:
The reasoning behind using `Consumer<SubBuilder>` is for users to have the ability to only override specific options in sub-builders of parent builders copied from another `FeatureSet` (via `FeatureSetBuilder#createFrom(FeatureSet)`, see [CSRG](https://github.com/NebelNidas/mapping-io/blob/6781380f85acd6c75728ff1ccb769e63b572bcbc/src/main/java/net/fabricmc/mappingio/format/MappingFormat.java#L175-L177) for example.

I'm not quite sure what exactly you're trying to suggest here, `FeatureSet build(Consumer<FeatureSetBuilder> featureApplier)` appears to me as a facility meant to enable chained builders, of whose usefulness I'm not very convinced